x/fswap/keeper/keeper.go

```diff
@@ -125,6 +131,14 @@ func (k Keeper) SetSwap(ctx sdk.Context, swap types.Swap, toDenomMetadata bank.M
 		return types.ErrCanNotHaveMoreSwap.Wrapf("cannot make more swaps, max swaps is %d", k.config.MaxSwaps)
 	}
 
+	if swap.ToDenom != toDenomMetadata.Base {
+		return sdkerrors.ErrInvalidRequest.Wrap("toDenom should be existed in metadata")
+	}
+
+	if !k.HasSupply(ctx, swap.FromDenom) {
```

Comment:
This brings another restriction to `SetSwap`.
This may not be necessary until we need this specification for a usecase.

Reply:
I think it is necessary to do so. This can prevent some misinputs and reduce the number of swaps in the array that will never be triggered. 
In my previous test, when only `stake` existed on the chain, the proposal about `cony` to `pdt` was passed. This was discovered during subsequent actual exchanges, but `cony` to `pdt` had already been recorded on the chain.

Reply:
@0Tech Since we currently do not have an interface that can detect unissued denom, I think we can ask users to do so. After all, this can prevent situations where only B->C is recorded, but A->B is not imported in the end.

Reply:
In my opinion, it's okay for fswap moudle to allow add `swap` first before other conditions are met such as there should be `from` metadata and there should be `FromDenom` supply.

Reply:
> In my opinion, it's okay for fswap moudle to allow add `swap` first before other conditions are met such as there should be `from` metadata and there should be `FromDenom` supply.

Checking `from` metadata has actually quite different restriction compared to checking supply. You can set swaps of (A->B), (B->C) with the metadata verification.

Reply:
Also, it would be a good idea to verify the existence of `from` metadata, because we are already checking `to` denom metadata in order to make sure the corresponding metadata must be registered.

Reply:
@jaeseung-bae I think it's necessary to `from_denom` should be issued first. Currently, `Swaps` that are set up through `foundation/proposals`. If the from_denom is not initially issued on the chain, then after the proposal is approved, the denom unit can be changed, thereby affecting the trading ratio.

Reply:
> If the from_denom is not initially issued on the chain, then after the proposal is approved, the denom unit can be changed, thereby affecting the trading ratio.

Sorry, I don't understand this part well.

I just feel like those validations are a little strict because they require a certain steps.
But It's okay to add.

Let me check what validation you want to add. 
1. Validation logic to check existence of denom-matadata for `from_denom`?
2. Validation logic to check supply?
3. Both 1 and 2.

Reply:
@jaeseung-bae 
Thank you for your feedback.
My main goal is to confirm point 1, perhaps I should use `GetDenomMetaData`? However, when I tried before, the `bondDenom` did not exist in the `DenomMetaData`, so I chose `HasSupply`... 
Maybe I should check both?

Reply:
Um, I don't have any particular preference.
As @0Tech said, since there is already one to check `to_denom` metadata, It's nice to add check `from_denom` metadata as well.

Reply:
I'm skeptical to add validation of 2. It will affect the UX.